app/models/shipit/stack.rb

```diff
@@ -74,7 +75,7 @@ def trigger_deploy(until_commit, user, env: nil)
         user_id: user.id,
         until_commit: until_commit,
         since_commit: since_commit,
-        env: env || {},
+        env: filter_deploy_envs(env || {}),
```

Comment:
I spent a long time contemplating where the filter should go and I'm still open to discussion on it. 

I could have filtered it in [`TaskCommands`](https://github.com/Shopify/shipit-engine/blob/add_env_params_api/lib/shipit/task_commands.rb#L30) but that would have restricted any sort of divergence from the whitelist